knotx-core/src/main/java/io/knotx/splitter/HtmlFragmentSplitter.java

```diff
@@ -35,14 +37,17 @@
 
   @Override
   public List<Fragment> split(String html) {
+    if (StringUtils.isEmpty(html)) {
+      throw new NoSuchElementException("html cannot be empty");
+    }
     List<Fragment> fragments = Lists.newLinkedList();
```

Comment:
I know that this is not part of the change you introduced, but could you change this to `ArrayList` here?
Thank you.

Reply:
done